app/models/full_text_search/oracle_searcher.rb

```diff
@@ -5,7 +5,14 @@ class OracleSearcher < BaseSearcher
     def search(fields, query)
       return @scope.none if query.blank?
 
-      formatted_query = query.to_s.split.join(",")
+      query_elements = query.to_s.split
+      # By wrapping each portion in {}, it escapes what's inside so it handles special
+      # characters and other reserved words
+      # https://docs.oracle.com/cd/B10501_01/text.920/a96518/cqspcl.htm#20741
+      formatted_query = query_elements
+        .map { |s| s.gsub(/[{}]/, "") } # remove any existing curly braces
+        .map { |s| "{#{s}}" }
```

Comment:
I'm guessing there aren't enough `query_elements` for this to make a performance impact, but you could do this all in one `map` like:
```
.map do |query_element| 
  query_string = query_element.gsub(/[{}]/, "") # remove any existing curly braces
  "{#{query_string}}"
end
```

Reply:
I think I prefer the multiple maps from a readability perspective; it's only a slight preference. If I were worried about the performance impact, I would definitely make the switch, though.